### ISSUE TITLE
Relation Extraction: A Survey[2017]

### ISSUE THREAD

**Issue by BrambleXu** [BrambleXu/knowledge-graph-learning#12]
# 一句话总结

这篇survey比之前 #10 那篇要更详细一些，有点长51页。

# 概要

## 2. Supervised Approaches
  - 2.1. Feature-based Methods： for each relation instance (i.e. pair of entity mentions) in the labelled data, a set of features is generated and a classiﬁer (or an ensemble of classiﬁers) is then trained to classify any new relation instance.
 - 2.2. Kernel Methods：因为基于feature的方法效果好坏完全看feature如何，而基于kernal的方法优点在于不用设计feature。kernel functions are designed to compute similarities between representations of two relation instances and SVM (Support Vector Machines) is employed for classiﬁcation. 这种方法的原理是计算relation实例之间的相似度，来进行分类。

## 3. Joint Extraction of Entities and Relations

第2章里全是基于feautre和kernel的监督式学习方法。但是如何事前没有标记好的POS，NER之类的标签的话，错误是会累计的。比如先预测NER出错，再判断relation仍然会出错。为了防止这种情况，于是把Entities and Relations的抽取任务结合起来处理。

- 3.1. Integer Linear Programming based Approach
- 3.2. Graphical Models based Approach
- 3.3. Card-Pyramid Parsing
- 3.4. Structured Prediction

## 4. Semi-supervised Approaches

Major motivation behind designing semi-supervised techniques is two-fold: i) to reduce the manual eﬀorts required to create labelled data; and ii) exploit the unlabelled data which is generally easily available without investing much eﬀorts.

- 4.1. Bootstrapping Approaches：bootstrapping algorithms require a large unlabelled corpus and a few seed instances of the relation type of interest.
- 4.2. Active Learning：The key idea behind active learning is that the learning algorithm is allowed to ask for true labels of some selected unlabelled instances. Various criterion have been proposed to choose these instances with the common objective of learning the underlying hypothesis quickly with a very few instances. The key advantage of active learning is that performance comparable with supervised methods is achieved through a very few labelled instances.
- 4.3 Label Propagation Method： Label Propagation is a graph based semi-supervised method proposed by Zhu and Ghahramani [130] where labelled and unlabelled instances in the data are represented as nodes in a graph with edges reﬂecting the similarity between nodes. In this method, the label information for any node is propagated to nearby nodes through weighted edges iteratively and ﬁnally the labels of unlabelled examples are inferred when the propagation process is converged.
- 4.4. Other Methods：Jiang [55] applied multi-task transfer learning to solve a weakly-supervised RE problem. 
- 4.5. Evaluation

## 5. Unsupervised Relation Extraction

- 5.1. Clustering based approaches
- 5.2. Other approaches

## 6. Open Information Extraction

1. Self-supervised Learner
2. Single Pass Extractor
3. Redundancy-based Assessor

## 7. Distant Supervision

Distant Supervision, proposed by Mintz et al. [75], is an alternative paradigm which does not require labelled data. The idea is to use a large semantic database for automatically obtaining relation type labels. Such labels may be noisy, but the huge amount of training data is expected to oﬀset this noise.

## 8. Recent Advances in Relation Extraction

- Universal Schemas
- n-ary Relation Extraction
- Cross-sentence Relation Extraction
- Convolutional Deep Neural Network
- Cross-lingual Annotation Projection
- Domain Adaptation

## 9. Conclusion and Future Research Directions

1. There have been several techniques for joint modelling of entity and relation extraction. However, the best reported F-measure on ACE 2004 dataset when gold-standard entities are not given, is still very low at around 48%. This almost 30% lower than the F-measure achieved when gold-standard entity information is assumed. Hence, there is still some scope of improvement here with more sophisticated models.

2. There has been little work for extracting n-ary relations, i.e. relations involving more than two entity mentions. There is a scope for more useful and principled approaches for this.

3. Most of the RE research has been carried out for English, followed by Chinese and Arabic, as ACE program released the datasets for these 3 languages. It would be interesting to analyse how eﬀective and language independent are the existing RE techniques. More systematic study is required for languages with poor resources (lack of good NLP pre-processing tools like POS taggers, parsers) and free word order, e.g. Indian languages.

4. Depth of the NLP processing used in most of the RE techniques, is mainly limited to lexical and syntax (constituency and dependency parsing) and few techniques use light semantic processing. It would be quite fruitful to analyse whether deeper NLP processing such as semantics and discourse level can help in improving RE performance.
